src/actions/index.js

```diff
@@ -0,0 +1,134 @@
+import api from '../api';
+import * as schema from './schema';
+import { asyncAction } from './utils';
+
+// Tasks
+export const fetchAllTasks = asyncAction({
+  type: 'tasks/FETCH_ALL_TASKS',
+  asyncCall: api.tofix.fetchAllTasks,
+  responseSchema: { tasks: schema.arrayOfTasks },
+});
+
+export const fetchTaskById = asyncAction({
+  type: 'tasks/FETCH_TASK_BY_ID',
+  asyncCall: api.tofix.fetchTaskById,
+  responseSchema: schema.task,
+});
+
+export const createTask = asyncAction({
+  type: 'tasks/CREATE_TASK',
+  asyncCall: api.tofix.createTask,
+  responseSchema: schema.task,
+});
+
+export const updateTask = asyncAction({
+  type: 'tasks/UPDATE_TASK',
+  asyncCall: api.tofix.updateTask,
+  responseSchema: schema.task,
+});
+
+export const deleteTask = asyncAction({
+  type: 'tasks/DELETE_TASK',
+  asyncCall: api.tofix.deleteTask,
+});
+
+export const setTaskId = ({ idtask }) => ({
+  type: 'tasks/SET_TASK_ID',
+  idtask,
+});
+
+// Items
+export const fetchAllItems = asyncAction({
+  type: 'items/FETCH_ALL_ITEMS',
+  asyncCall: api.tofix.fetchAllItems,
+  responseSchema: schema.arrayOfItems,
+});
+
+export const fetchRandomItem = asyncAction({
+  type: 'items/FETCH_RANDOM_ITEM',
+  asyncCall: api.tofix.fetchRandomItem,
+  responseSchema: schema.item,
+});
+
+export const fetchItemByKey = asyncAction({
+  type: 'items/FETCH_ITEM_BY_KEY',
+  asyncCall: api.tofix.fetchItemByKey,
+  responseSchema: schema.item,
+});
+
+export const fetchNItems = asyncAction({
+  type: 'items/FETCH_N_ITEMS',
+  asyncCall: api.tofix.fetchNItems,
+  responseSchema: schema.arrayOfItems,
+});
+
+export const updateItem = asyncAction({
+  type: 'items/UPDATE_ITEM',
+  asyncCall: api.tofix.updateItem,
+});
+
+export const unlockItems = asyncAction({
+  type: 'items/UNLOCK_ITEMS',
+  asyncCall: api.tofix.unlockItems,
+});
+
+// Activity
+export const fetchActivity = asyncAction({
+  type: 'activity/FETCH_ACTIVITY',
+  asyncCall: api.tofix.fetchActivity,
+});
+
+export const fetchUserActivity = asyncAction({
+  type: 'activity/FETCH_USER_ACTIVITY',
+  asyncCall: api.tofix.fetchUserActivity,
+});
+
+// Stats
+export const fetchStats = asyncAction({
+  type: 'stats/FETCH_STATS',
+  asyncCall: api.tofix.fetchStats,
+});
+
+// User
+export const login = asyncAction({
+  type: 'user/LOGIN',
+  asyncCall: api.osm.login,
+});
+
+export const getUserDetails = asyncAction({
+  type: 'user/GET_USER_DETAILS',
+  asyncCall: api.osm.getUserDetails,
+});
+
+export const logout = () => (dispatch) => {
+  api.osm.logout();
+  dispatch({ type: 'user/LOGOUT' });
+};
+
+// Settings
+export const toggleSidebar = () => ({
+  type: 'settings/TOGGLE_SIDEBAR'
```

Comment:
action types are a good thing to store as constants, since they mean that you're relying on an exact string value at two places - action construction (here) and in the store, so it's a place where strings can quietly get refactored out of correspondence and cause the application to fail silently